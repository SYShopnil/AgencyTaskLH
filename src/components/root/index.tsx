import Button from './button';

export { Button };
export { BannerSectionBar } from './banner-bar';
export { SlideCard } from './slide-card/';
export { HeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition';
export { Navbar } from './navbar';
