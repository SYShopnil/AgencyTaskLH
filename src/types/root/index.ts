export type { IButton, INormalButton, IRedirectButton } from './Button/';

export { BtnColorSchema } from './Button/';

export type { IBtnCard } from './btn-card';
export { ColorSchema, CardSize } from './btn-card';

export type { IIconStore } from './_icon';
export { IconName as IconEnums } from './_icon';

export type { ICurveFrame } from './curveFrame';
export { CurveFrameVariantEnum } from './curveFrame';

export type { IInfoCard } from './infoCard';
export type { ITitleDescSection } from './title_desc_sec/';

export type { IBanner, IBannerSectionBar } from './banner-bar/banner.ts';
export type { ISlideCard } from './slide-card';

export type { IHeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition/index';
export type { IFooter } from './footer';

export type {
  ICommonSection,
  IMegaMenu,
  IMegaMenuItems,
  ISubMenuBlock,
  INavbar,
} from './navbar';
