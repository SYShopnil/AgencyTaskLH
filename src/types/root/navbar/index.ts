import { IconEnums } from '..';

export interface ISubMenuBlock {
  iconName: IconEnums;
  title: string;
  subTitle: string;
  link: string;
}

export interface IMegaMenuItems {
  heading: string;
  subHeading: string;
  contents: ISubMenuBlock[];
}

export interface ICommonSection {
  logo: IconEnums.Logo;
  title: string;
  redirectLInk: string;
}

export interface IMegaMenu {
  menuItems: IMegaMenuItems[];
  commonPart: ICommonSection;
}

interface INavbarItems {
  heading: string;
  megaMenuItems: IMegaMenuItems[];
}
export interface INavbar {
  navItems: INavbarItems[];
  commonPart: ICommonSection;
  getInTouchLink: string;
  dataTestId?: string;
}
