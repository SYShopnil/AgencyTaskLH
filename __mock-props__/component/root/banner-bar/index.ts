import { IBanner, IBannerSectionBar } from '@src/types/root';
import { IconName } from '@src/types/root/_icon';

export const BannerImageMockData: IBanner[] = [
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
];

export const BannerBarSectionMockData: IBannerSectionBar = {
  contents: BannerImageMockData,
  bannerBarSectionTestId: 'mainWrapper',
};
