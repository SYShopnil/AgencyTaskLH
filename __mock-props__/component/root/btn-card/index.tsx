import { CardSize, ColorSchema, IBtnCard } from '@src/types/root';

export const btnCardMockProps: IBtnCard[] = [
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_lg,
    fill: 65,
    data_test_id: 'btnCardOne',
  },
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_sm,
    fill: 50,
    data_test_id: 'btnCardTwo',
  },
];
