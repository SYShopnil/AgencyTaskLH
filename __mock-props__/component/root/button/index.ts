import { BtnColorSchema, IRedirectButton } from '@src/types/root';

export const mock_data_redirectButton: IRedirectButton[] = [
  {
    btnText: 'Back to Homepage',
    colorSchema: BtnColorSchema.BlackBgTextWhite,
    isOpenNewTab: true,
    redirectLink: 'https://chat.openai.com/',
    dataTestId: 'Redirect Button One',
  },
  {
    btnText: 'Back to Homepage Two',
    colorSchema: BtnColorSchema.BlackBgTextWhite,
    isOpenNewTab: false,
    redirectLink: '',
    dataTestId: 'Redirect Button Two',
  },
];
