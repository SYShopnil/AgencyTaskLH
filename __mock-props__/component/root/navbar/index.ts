import { INavbar, ISubMenuBlock, IconEnums } from '@src/types/root';

const menuItemsDummyData: ISubMenuBlock[] = [
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable web apps',
    title: 'Progressive Web Apps',
  },
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable web apps',
    title: 'Artificial Intelligence',
  },
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable web apps',
    title: 'Progressive Web Apps',
  },
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable web apps',
    title: 'Artificial Intelligence',
  },
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable web apps',
    title: 'Progressive Web Apps',
  },
];

const menuItemsDummyDataTwo: ISubMenuBlock[] = [
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable Mobile apps',
    title: 'Progressive Mobile Apps',
  },
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable Web apps',
    title: 'Progressive Mobile Apps',
  },
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable Mobile apps',
    title: 'Progressive Mobile Apps',
  },
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable Mobile apps',
    title: 'Progressive Mobile Apps',
  },
  {
    iconName: IconEnums.NavIconOne,
    link: 'https://www.facebook.com/',
    subTitle: 'Offline-capable Mobile apps',
    title: 'Progressive Mobile Apps',
  },
];

export const navBarMockData: INavbar = {
  commonPart: {
    logo: IconEnums.Logo,
    redirectLInk: 'https://www.facebook.com/',
    title: `The Best Design Agency`,
  },
  navItems: [
    {
      heading: 'Home',
      megaMenuItems: [
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyData,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyDataTwo,
        },
        {
          heading: 'Headless Web App',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyData,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyDataTwo,
        },
      ],
    },
    {
      heading: 'Services',
      megaMenuItems: [
        {
          heading: 'Hello website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyData,
        },
        {
          heading: 'Headless Application',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyDataTwo,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyData,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyDataTwo,
        },
      ],
    },
    {
      heading: 'Projects',
      megaMenuItems: [
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyData,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyDataTwo,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyData,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyDataTwo,
        },
      ],
    },
    {
      heading: 'Blogs',
      megaMenuItems: [
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyData,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyDataTwo,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyData,
        },
        {
          heading: 'Headless website',
          subHeading: 'Build modern, decoupled web experiences',
          contents: menuItemsDummyDataTwo,
        },
      ],
    },
  ],
  getInTouchLink: 'https://www.facebook.com/',
  dataTestId: 'navBarCatcher',
};
