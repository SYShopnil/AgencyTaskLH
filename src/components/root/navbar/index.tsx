import React, { useState } from 'react';
import { MegaMenu } from './megaMenu';
import { BtnColorSchema, IMegaMenuItems, INavbar } from '@src/types/root';
import Button from '../button';
import ModuleStyle from './index.module.scss';

export const Navbar = ({
  commonPart,
  navItems,
  getInTouchLink,
  dataTestId,
}: INavbar) => {
  const [activeItem, setActiveItem] = useState<number>(-1);
  const mouseOnEnterNavItemHandler = (e: any, ind: number) => {
    e.preventDefault();
    setActiveItem(ind);
  };

  const mouseOnLeaveNavItemHandler = (
    e: React.MouseEvent<HTMLDivElement, MouseEvent>
  ) => {
    e.preventDefault();
    setActiveItem(-1);
  };

  return (
    <nav data-testid={dataTestId || 'navBarCatcher'}>
      {/* nav bar wrapper */}
      <div
        className={`space-y-[1.375rem] relative h-[12rem] min-w-[48.75rem]`}
        onMouseLeave={(e) => mouseOnLeaveNavItemHandler(e)}
      >
        <ul className={`grid grid-cols-10 h-[3.5rem] space-x-[4.5rem]`}>
          {navItems.map((navItem, ind) => {
            return (
              <li
                key={ind}
                className={`relative col-span-2  flex justify-center items-center cursor-pointer`}
                onMouseEnter={(e) => mouseOnEnterNavItemHandler(e, ind)}
              >
                <a
                  className={`text-[1.25rem] ${
                    activeItem == ind
                      ? ModuleStyle['active']
                      : ModuleStyle['non-active']
                  }`}
                >
                  {navItem.heading}
                </a>
                <div
                  className={`${ModuleStyle['activeDot']} ${
                    activeItem == ind && ModuleStyle['show']
                  }`}
                ></div>
              </li>
            );
          })}
          <li className={`col-span-2`}>
            <span>
              <Button.RedirectButton
                btnText="Get In Touch"
                colorSchema={BtnColorSchema.TransparentBgTextPrimary500}
                isArrow={false}
                isOpenNewTab={true}
                redirectLink={getInTouchLink}
              />
            </span>
          </li>
        </ul>
        {/* mega menu wrapper */}
        <div className={`${ModuleStyle['setMegaMenuBarPosition']}`}>
          {navItems.map((navItem, ind) => {
            const { megaMenuItems } = navItem;
            return (
              <>
                {activeItem == ind && (
                  <div className={``}>
                    <MegaMenu
                      commonPart={commonPart}
                      menuItems={megaMenuItems}
                    />
                  </div>
                )}
              </>
            );
          })}
        </div>
      </div>
    </nav>
  );
};
