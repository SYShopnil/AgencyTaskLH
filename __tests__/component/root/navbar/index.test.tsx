import { navBarMockData } from '@root/__mock-props__/component/root/navbar';
import { Navbar } from '@src/components/root';
import { render } from '@testing-library/react';

describe('When Navbar has been loaded', () => {
  it('Expect it will render successfully', () => {
    const { getByTestId } = render(<Navbar {...navBarMockData} />);
    const catchTheCard = getByTestId(
      navBarMockData?.dataTestId || 'navBarCatcher'
    );
    expect(catchTheCard).toBeInTheDocument();
    expect(catchTheCard).toMatchSnapshot();
  });
});
