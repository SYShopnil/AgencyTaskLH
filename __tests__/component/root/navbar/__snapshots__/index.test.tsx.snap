// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Navbar has been loaded Expect it will render successfully 1`] = `
<nav
  data-testid="navBarCatcher"
>
  <div
    class="space-y-[1.375rem] relative h-[12rem] min-w-[48.75rem]"
  >
    <ul
      class="grid grid-cols-10 h-[3.5rem] space-x-[4.5rem]"
    >
      <li
        class="relative col-span-2  flex justify-center items-center cursor-pointer"
      >
        <a
          class="text-[1.25rem] non-active"
        >
          Home
        </a>
        <div
          class="activeDot false"
        />
      </li>
      <li
        class="relative col-span-2  flex justify-center items-center cursor-pointer"
      >
        <a
          class="text-[1.25rem] non-active"
        >
          Services
        </a>
        <div
          class="activeDot false"
        />
      </li>
      <li
        class="relative col-span-2  flex justify-center items-center cursor-pointer"
      >
        <a
          class="text-[1.25rem] non-active"
        >
          Projects
        </a>
        <div
          class="activeDot false"
        />
      </li>
      <li
        class="relative col-span-2  flex justify-center items-center cursor-pointer"
      >
        <a
          class="text-[1.25rem] non-active"
        >
          Blogs
        </a>
        <div
          class="activeDot false"
        />
      </li>
      <li
        class="col-span-2"
      >
        <span>
          <a
            href="https://www.facebook.com/"
            target="_blank"
          >
            <span
              class="flex justify-center items-center  "
            >
              <span
                class="text-base_2  flex justify-center items-center px-[1.5rem] py-[1rem] rounded-primary  bg-[transparent] hover:bg-primary_500 duration-[0.5s] text-primary_500 hover:text-neutral_500 border-[0.063rem] border-primary_500  w-[100%] h-[100%]"
              >
                <span>
                  Get In Touch
                </span>
              </span>
            </span>
          </a>
        </span>
      </li>
    </ul>
    <div
      class="setMegaMenuBarPosition"
    />
  </div>
</nav>
`;
